Taken from Oracle solaris-userland b83a793,
components/x11/xserver/xorg/patches/08-IA.patch (xorg-server 21.1.24).

Solaris-specific support for the SolarisIA extension

Not suitable for upstream

--- xorg-server-21.1.20/Makefile.am
+++ xorg-server-21.1.20/Makefile.am
@@ -58,6 +58,7 @@ $(COMPOSITE_DIR) \
 	$(GLX_DIR) \
 	$(PRESENT_DIR) \
 	$(DRI3_DIR) \
+	IA \
 	exa \
 	$(GLAMOR_DIR) \
 	config \
--- xorg-server-21.1.20/configure.ac
+++ xorg-server-21.1.20/configure.ac
@@ -1341,6 +1341,10 @@ AC_DEFINE(XINPUT, 1, [Support X Input ex
 XI_LIB='$(top_builddir)/Xi/libXi.la'
 XI_INC='-I$(top_srcdir)/Xi'
 
+# SolarisIA extension
+IA_LIB='$(top_builddir)/IA/libIA.la'
+XEXT_LIB="${XEXT_LIB} ${IA_LIB}"
+
 AM_CONDITIONAL(XF86UTILS, test "x$XF86UTILS" = xyes)
 AM_CONDITIONAL(VGAHW, test "x$VGAHW" = xyes)
 AM_CONDITIONAL(INT10MODULE, test "x$INT10MODULE" = xyes)
@@ -2385,3 +2389,6 @@ xserver.ent
 xorg-server.pc
 ])
 AC_OUTPUT
+
+# Add Sun IA extension
+AC_OUTPUT([IA/Makefile])
--- xorg-server-21.1.20/include/extinit.h
+++ xorg-server-21.1.20/include/extinit.h
@@ -115,6 +115,14 @@ extern _X_EXPORT Bool noMITShmExtension;
 extern void ShmExtensionInit(void);
 #endif
 
+#define SolarisIAExtension
+
+#ifdef SolarisIAExtension
+#include <X11/extensions/interactive.h>
+extern _X_EXPORT Bool noIAExtension;
+extern void IAExtensionInit(void);
+#endif
+
 extern void SyncExtensionInit(void);
 
 extern void XCMiscExtensionInit(void);
--- xorg-server-21.1.20/mi/miinitext.c
+++ xorg-server-21.1.20/mi/miinitext.c
@@ -167,6 +167,9 @@ #ifdef XV
     {XvExtensionInit, "XVideo", &noXvExtension},
     {XvMCExtensionInit, "XVideo-MotionCompensation", &noXvExtension},
 #endif
+#ifdef SolarisIAExtension
+    {IAExtensionInit, "SolarisIA", &noIAExtension},
+#endif
 #ifdef XSELINUX
     {SELinuxExtensionInit, "SELinux", &noSELinuxExtension},
 #endif
--- xorg-server-21.1.20/os/utils.c
+++ xorg-server-21.1.20/os/utils.c
@@ -177,6 +177,9 @@ #endif
 #ifdef DRI2
 Bool noDRI2Extension = FALSE;
 #endif
+#ifdef SolarisIAExtension
+Bool noIAExtension = FALSE;
+#endif
 
 Bool noGEExtension = FALSE;
 
